apps/meteor/server/models/raw/Messages.ts

```diff
@@ -1362,22 +1363,28 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 			return count;
 		}
 
-		const messagesToDelete = (
-			await this.find(query, {
-				projection: {
-					_id: 1,
-				},
-				limit,
-			}).toArray()
-		).map(({ _id }) => _id);
+		const messagesToDelete = await this.find(query, {
+			projection: {
+				_id: 1,
+				_hidden: 1,
+				t: 1,
+				editedAt: 1,
+				editedBy: 1,
+			},
+			limit,
+		}).toArray();
 
-		const count = (
-			await this.deleteMany({
-				_id: {
-					$in: messagesToDelete,
-				},
-			})
-		).deletedCount;
+		const notCountedMessages = messagesToDelete.filter((message) => message._hidden === true || isDeletedMessage(message)).length;
```

Comment:
Wouldn't it be better to filter this at the DB level? Or even use only the `deleteMany` method, without querying the message first (not sure if it's possible 🤔 )

Reply:
Done! I think using only `deleteMany` isn't possible due to the `limit` param :thinking: